src/engine.jl

```diff
@@ -40,53 +36,49 @@ type MSession
             # Hide the MATLAB Command Window on Windows
             ccall(engfunc(:engSetVisible ), Cint, (Ptr{Void}, Cint), ep, 0)
         end
-
-        println("A MATLAB session is open successfully")
         new(ep, buf, bufptr)
     end
-
-    MSession() = MSession(default_output_buffer_size)
 end
 
+
 function close(session::MSession)
     # Close a MATLAB Engine session
-
     @assert libeng::Ptr{Void} != C_NULL
-
     r = ccall(engfunc(:engClose), Cint, (Ptr{Void},), session.ptr)
-    if r != 0
-        throw(MEngineError("Failed to close a MATLAB engine session (err = $r)"))
-    end
+    r != 0 && throw(MEngineError("Failed to close a MATLAB engine session (err = $r)"))
+    session.ptr = C_NULL
+    return nothing
 end
 
 # default session
 
 default_msession = nothing
 
-function restart_default_msession(bufsize::Integer)
+function restart_default_msession(bufsize::Int = default_output_buffer_size)
     global default_msession
-    if !(default_msession == nothing)
+    if default_msession != nothing
         close(default_msession)
     end
     default_msession = MSession(bufsize)
+    return nothing
 end
 
-restart_default_msession() = restart_default_msession(default_output_buffer_size)
 
 function get_default_msession()
     global default_msession
     if default_msession == nothing
         default_msession = MSession()
     end
-    default_msession::MSession
+    return default_msession
```

Comment:
oh nvm, didnt ' realize the inference isn't able to pick this one up. ok thanks.